osf/utils/machines.py

```diff
@@ -97,7 +98,66 @@ def update_last_transitioned(self, ev):
         self.machineable.date_last_transitioned = now
 
 
-class ReviewsMachine(BaseMachine):
+class ModerationNotificationMixin(BaseMachine):
```

Comment:
Driveby: This really doesn't need to be a Machine subclass. All you're gaining is direct access to "machineable" and saving a step by not having to write your state machine callbacks.

In general, I'd say this is cleaner as a helper class (or just util module?) that generates and sends emails given a registration/preprint and trigger